tests/conftest.py

```diff
@@ -41,3 +45,43 @@ def cdn_test_url(request):
     else:
         pytest.skip("Test skipped without --cdn-test-url or --lambda-stack")
     return url
+
+
+def mock_conf_file():
+    temp_file = tempfile.NamedTemporaryFile(
+        prefix="lambda_unitests_", delete=False
+    )
+
+    os.environ["EXODUS_LAMBDA_CONF_FILE"] = temp_file.name
+
+    test_env = os.environ.copy()
+
+    test_env["PROJECT"] = "test"
+    test_env["ENV_TYPE"] = "dev"
+    test_env["EXODUS_CONFIG_CACHE_TTL"] = "2"
+    test_env["ORIGIN_RESPONSE_LOGGER_LEVEL"] = "DEBUG"
+    test_env["ORIGIN_REQUEST_LOGGER_LEVEL"] = "DEBUG"
+    test_env["EXODUS_HEADERS_MAX_AGE"] = "600"
+
+    cmd = "envsubst < ./configuration/lambda_config.json > {temp_path}"
+    cmd = cmd.format(temp_path=temp_file.name)
+
+    subprocess.run(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=True,
+        shell=True,
+        env=test_env,
+    )
+
+    return temp_file
+
+
+TEMP_FILE = mock_conf_file()
+
+
+def pytest_sessionfinish(session, exitstatus):
+    # remove temp conf after whole test run finished
```

Comment:
Minor - instead of creating the file with delete=False and then having to both close and delete it here, how about removing the delete=False ?  Then you just need to close the file and it will be deleted.